docs/extended_api/pipelines/pipeline_shared_state.md

```diff
@@ -0,0 +1,82 @@
+---
+grand_parent: Extended API
+parent: Pipelines
+---
+
+# `cuda::pipeline_shared_state`
+
+Defined in header `<cuda/pipeline>`:
+
+```cuda
+template <cuda::thread_scope Scope, cuda::std::uint8_t StagesCount>
+class cuda::pipeline_shared_state {
+public:
+  __host__ __device__
+  pipeline_shared_state();
+
+  ~pipeline_shared_state() = default;
+
+  pipeline_shared_state(pipeline_shared_state const&) = delete;
+
+  pipeline_shared_state(pipeline_shared_state&&) = delete;
+};
+```
+
+The class template `cuda::pipeline_shared_state` is a storage type used to
+  coordinate the threads participating in a `cuda::pipeline`.
+
+## Template Parameters
+
+| `Scope`       | A [`cuda::thread_scope`] denoting a scope including all the threads participating in the `cuda::pipeline`. `Scope` cannot be `cuda::thread_scope_thread`. |
+| `StagesCount` | The number of stages for the _pipeline_.                                                                                                                  |
+
+## Member Functions
+
+| [(constructor)]                    | Constructs a `cuda::pipeline_shared_state`.      |
+| (destructor) [implicitly declared] | Destroys the `cuda::pipeline_shared_state`. |
+| `operator=` [deleted]              | `cuda::pipeline_shared_state` is not assignable. |
+
+## NVCC `__shared__` Initialization Warnings
```

Comment:
Since we have this in many places maybe we should just factorize it into a separate `.md` and just include it.

Reply:
Yah, that may not be a bad idea; I don't think we have to do it in this change though.